### ISSUE TITLE
Weird warnings in covidcast_signals() when specifying as_of but no end_day

### ISSUE THREAD

**Issue by ryantibs** [cmu-delphi/covidcast#390]
I'm using the version of covidcast up on the r/0.4.0-release branch.  When I use `covidcast_signal()` or the plural form, I should be able to specify a `start_day`, `as_of`, and *no* `end_day` parameter and it should be fetch to pull everything that was available up to that `as_of` date.  This is just extending the expected default behavior of this function when `as_of` is set to the current day (left as `NULL`) to the case when `as_of` is explicitly set to be sometime back in the past.  But try running:

```
forecast_date = as.Date("2020-07-10")
df = suppressMessages(
  covidcast_signals(data_source = "usa-facts", 
                       signal = "confirmed_7dav_incidence_prop",
                       start_day = forecast_date - 10,
                       as_of = forecast_date, geo_type = "state"))
```

You'll see you get a bunch of weird warnings.

```
Warning messages:
1: In rep(end_day, length.out = N) :
  'x' is NULL so the result will be NULL
2: Data not fetched for the following days: 2020-07-09, 2020-07-10, 2020-07-11, 2020-07-12, 2020-07-13, 2020-07-14, 2020-07-15, 2020-07-16, 2020-07-17, 2020-07-18, 2020-07-19, 2020-07-20, 2020-07-21, 2020-07-22, 2020-07-23, 2020-07-24, 2020-07-25, 2020-07-26, 2020-07-27, 2020-07-28, 2020-07-29, 2020-07-30, 2020-07-31, 2020-08-01, 2020-08-02, 2020-08-03, 2020-08-04, 2020-08-05, 2020-08-06, 2020-08-07, 2020-08-08, 2020-08-09, 2020-08-10, 2020-08-11, 2020-08-12, 2020-08-13, 2020-08-14, 2020-08-15, 2020-08-16, 2020-08-17, 2020-08-18, 2020-08-19, 2020-08-20, 2020-08-21, 2020-08-22, 2020-08-23, 2020-08-24, 2020-08-25, 2020-08-26, 2020-08-27, 2020-08-28, 2020-08-29, 2020-08-30, 2020-08-31, 2020-09-01, 2020-09-02, 2020-09-03, 2020-09-04, 2020-09-05, 2020-09-06, 2020-09-07, 2020-09-08 
3: Fetching confirmed_7dav_incidence_prop from usa-facts for 2020-09-09 to 2020-11-18 in geography '*': no results 
4: Fetching confirmed_7dav_incidence_prop from usa-facts for 2020-11-19 to 2020-12-24 in geography '*': no results 
```

This first warning is specific to `covidcast_signals()` plural form and can be fixed just by modifying [this line](https://github.com/cmu-delphi/covidcast/blob/015fc0189b47befad5b4bf802cec9f83fee59d6c/R-packages/covidcast/R/covidcast.R#L530) and similarly the previous line to protect against the case when one of `start_day`, `end_day` is `NULL`.  

The second through fourth warnings are due to (what looks like) an attempt to fetch data at dates *past* the `as_of` date, which is obviously wrong. I'm guessing that since `end_day` is `NULL`, it's being set from the metadata as the last available date, and then it's throwing warnings it's not able to retrieve this data once we set `as_of`.  We should find a way to avoid these warnings, which are unnecessary and confusing.

**Comment by capnrefsmmat** [comment]
> The second through fourth warnings are due to (what looks like) an attempt to fetch data at dates past the as_of date, which is obviously wrong. I'm guessing that since end_day is NULL, it's being set from the metadata as the last available date, and then it's throwing warnings it's not able to retrieve this data once we set as_of. We should find a way to avoid these warnings, which are unnecessary and confusing.

This is almost certainly exactly what's happening, and we could avoid it by using `min(as_of, last_day_from_metadata)` as the last date to query.

**Comment by krivard** [comment]
Fixed in #394.
